packages/runtime-html/src/attribute-syntax-transformer.ts

```diff
@@ -1,17 +1,77 @@
 import { DI } from '@aurelia/kernel';
-import { AttrSyntax } from './resources/attribute-pattern.js';
+import type { AttrSyntax } from './resources/attribute-pattern.js';
 
 export interface IAttrSyntaxTransformer extends AttrSyntaxTransformer {}
-export const IAttrSyntaxTransformer = DI.createInterface<IAttrSyntaxTransformer>('IAttrSyntaxTransformer').withDefault(x => x.singleton(AttrSyntaxTransformer));
+export const IAttrSyntaxTransformer = DI
+  .createInterface<IAttrSyntaxTransformer>('IAttrSyntaxTransformer')
+  .withDefault(x => x.singleton(AttrSyntaxTransformer));
+
+type ITwoWayTransformerFn = (element: HTMLElement, property: PropertyKey) => boolean;
 
 export class AttrSyntaxTransformer {
+  /**
+   * @internal
+   */
+  private readonly fns: ITwoWayTransformerFn[] = [];
+  public constructor() {
+    this.useTwoWay((element, property) => {
+      switch (element.tagName) {
+        case 'INPUT':
+          switch ((element as HTMLInputElement).type) {
+            case 'checkbox':
+            case 'radio':
+              return property === 'checked';
+            default:
+              return property === 'value' || property === 'files';
+          }
+        case 'TEXTAREA':
+        case 'SELECT':
+          return property === 'value';
+        default:
+          switch (property) {
+            case 'textContent':
+            case 'innerHTML':
+              return element.hasAttribute('contenteditable');
+            case 'scrollTop':
+            case 'scrollLeft':
+              return true;
+            default:
+              return false;
+          }
+      }
+    });
+  }
+
+  /**
+   * Add a given function to a list of fns that will be used
+   * to check if `'bind'` command can be transformed to `'two-way'` command.
+   *
+   * If one of those functions in this lists returns true, the `'bind'` command
+   * will be transformed into `'two-way'` command.
+   *
+   * The function will be called with 2 parameters:
+   * - element: the element that the template compiler is currently working with
+   * - property: the target property name
+   */
+  public useTwoWay(fn: ITwoWayTransformerFn): void {
+    this.fns.push(fn);
+  }
+
+  /**
+   * @internal
+   */
   public transform(node: HTMLElement, attrSyntax: AttrSyntax): void {
-    if (attrSyntax.command === 'bind' && shouldDefaultToTwoWay(node, attrSyntax)) {
+    attrSyntax.target = this.map(node.tagName, attrSyntax.target);
+    if (attrSyntax.command === 'bind' && this.fns.some(fn => fn(node, attrSyntax.target))) {
```

Comment:
```suggestion
    if (
      attrSyntax.command === 'bind' &&
      (
        shouldDefaultToTwoWay(node, attrSyntax) ||
        this.fns.length > 0 && this.fns.some(fn => fn(node, attrSyntax.target))
      )
    ) {
```
Should only use the `fns` array if there are actually functions registered, otherwise we get an unnecessary slowdown in  the common case.